tensorcircuit/applications/optimization.py

```diff
@@ -170,17 +169,17 @@ def cvar_value(r: List[float], p: List[float], percent: float) -> float:
     :param percent: The cut-off percentage of CVaR.
     :return: The calculated CVaR value.
     """
-    sorted_indices = np.argsort(r)
-    p = np.array(p)[sorted_indices]  # type: ignore
-    r = np.array(r)[sorted_indices]  # type: ignore
+    sorted_indices = tf.argsort(r)
+    p = tf.cast(tf.gather(p, sorted_indices), dtype=tf.float32)
+    r = tf.cast(tf.gather(r, sorted_indices), dtype=tf.float32)
 
-    sump = 0.0  # The sum of probabilities.
-    count = 0
-    cvar_result = 0.0
+    sump = tf.constant(0.0, dtype=tf.float32)  # The sum of probabilities.
+    count = tf.constant(0, dtype=tf.int32)
+    cvar_result = tf.constant(0.0, dtype=tf.float32)
 
     # Iterate over the sorted results and calculate CVaR.
     while sump < percent:
-        if round(sump + p[count], 6) >= percent:
+        if tf.math.round(sump + p[count], 6) >= percent:
```

Comment:
concerned on whether this is really jittable, due to the ``if tensor`` pattern here, where ``tensor`` directly depends on the jittable input argument of the function

Reply:
I think the correct way is to build a sign function depending on the threshold to filter the results with higher loss without using the ``if tensor`` pattern